diesel/src/pg/connection/row.rs

```diff
@@ -20,10 +20,10 @@ impl<'a> PgRow<'a> {
 }
 
 impl<'a> Row<Pg> for PgRow<'a> {
-    fn take(&mut self) -> Option<&[u8]> {
+    fn take(&mut self) -> Option<PgValue> {
         let current_idx = self.col_idx;
         self.col_idx += 1;
-        self.db_result.get(self.row_idx, current_idx)
+        self.db_result.get(self.row_idx, current_idx).map(|result| result)
```

Comment:
This map call seems to be redundant.

```suggestion
        self.db_result.get(self.row_idx, current_idx)
```